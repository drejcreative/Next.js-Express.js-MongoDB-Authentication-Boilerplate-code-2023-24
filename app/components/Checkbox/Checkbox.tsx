import React, { useState, useEffect } from "react";

import style from "./Checkbox.module.scss";

interface Props {
  onChange: (val: any) => any;
  label: string;
  value: boolean;
  link?: boolean;
  className?: string;
  name: string;
}

const Checkbox = ({ onChange, label, link, value, name, className }: Props) => {
  const [checked, setChecked] = useState<boolean>(false);

  useEffect(() => {
    setChecked(value);
  }, [value]);

  const handleCheckboxChange = (e: any) => {
    setChecked(e.target.checked);
    e.target.value = e.target.checked;
    onChange(e);
  };

  return (
    <div className={`${style.checkbox} ${className}`}>
      <input type="checkbox" checked={!!checked} name={name} onChange={handleCheckboxChange} />
      {label &&
        (link ? (
          <a target="_blank" href="/terms" rel="noopener noreferrer">
            <label style={{ cursor: "pointer" }}>{label}</label>
          </a>
        ) : (
          <label>{label}</label>
        ))}
    </div>
  );
};

export default Checkbox;
