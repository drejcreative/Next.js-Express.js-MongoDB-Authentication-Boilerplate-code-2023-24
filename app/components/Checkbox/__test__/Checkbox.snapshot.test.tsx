import React from "react";
import { render } from "@testing-library/react";

import Checkbox from "../Checkbox";

it("Checkbox", () => {
  const { container } = render(
    <Checkbox label="test" name="checkbox" value="true" onChange={() => {}} />
  );
  expect(container).toMatchSnapshot();
});

it("Checkbox with empty value", () => {
  const { container } = render(
    <Checkbox label="test" name="checkbox" value="" onChange={() => {}} />
  );
  expect(container).toMatchSnapshot();
});
