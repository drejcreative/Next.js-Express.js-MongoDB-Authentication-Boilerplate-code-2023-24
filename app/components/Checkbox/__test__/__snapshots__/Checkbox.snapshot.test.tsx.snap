// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox 1`] = `
<div>
  <div
    class="checkbox undefined"
  >
    <input
      name="checkbox"
      type="checkbox"
    />
    <label>
      test
    </label>
  </div>
</div>
`;

exports[`Checkbox with empty value 1`] = `
<div>
  <div
    class="checkbox undefined"
  >
    <input
      name="checkbox"
      type="checkbox"
    />
    <label>
      test
    </label>
  </div>
</div>
`;
